myapp/app/models/task.rb

```diff
@@ -9,12 +9,27 @@ class Task < ApplicationRecord
   validate :due_cannot_be_earlier_than_now, if: :due_changed?
 
   belongs_to :user
+  has_many :tasks_labels, dependent: :destroy
+  has_many :labels, through: :tasks_labels
 
   def self.search_with_sort(query_title, query_status, sort_by, order)
     if query_status.empty?
```

Comment:
[FYI]
It might be simpler like this. This is just sharing. No need to fix.
```
tasks = where('title LIKE ?', "%#{query_title}%")
tasks = tasks.where(status: query_status) if query_status.present?
tasks.order("tasks.#{sort_by} #{order}")
```